tests/test_config.py

```diff
@@ -186,13 +186,15 @@ def test_set_config_value_file_permissions(self):
         file_mode = os.stat(self.cli_config.config_path).st_mode
         self.assertTrue(bool(file_mode & stat.S_IRUSR))
         self.assertTrue(bool(file_mode & stat.S_IWUSR))
-        self.assertFalse(bool(file_mode & stat.S_IXUSR))
-        self.assertFalse(bool(file_mode & stat.S_IRGRP))
-        self.assertFalse(bool(file_mode & stat.S_IWGRP))
-        self.assertFalse(bool(file_mode & stat.S_IXGRP))
-        self.assertFalse(bool(file_mode & stat.S_IROTH))
-        self.assertFalse(bool(file_mode & stat.S_IWOTH))
-        self.assertFalse(bool(file_mode & stat.S_IXOTH))
+        # only S_IRUSR and S_IWUSR are supported on Windows: https://docs.python.org/3.8/library/os.html#os.chmod
+        if os.name != 'nt':
+            self.assertFalse(bool(file_mode & stat.S_IXUSR))
+            self.assertFalse(bool(file_mode & stat.S_IRGRP))
+            self.assertFalse(bool(file_mode & stat.S_IWGRP))
+            self.assertFalse(bool(file_mode & stat.S_IXGRP))
+            self.assertFalse(bool(file_mode & stat.S_IROTH))
+            self.assertFalse(bool(file_mode & stat.S_IWOTH))
+            self.assertFalse(bool(file_mode & stat.S_IXOTH))
```

Comment:
Windows doesn't have Linux-like user/group/other concepts for `stat`. `rwx` is always the same for all scopes.

Reply:
![image](https://user-images.githubusercontent.com/14357159/74915125-a0a22180-53fe-11ea-8868-419ea63435eb.png)

```
>>> import sys
>>> sys.version
'3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)]'
```

As far as I understand from the doc, the limitation only takes effect on `os.chmod`, however, `os.stat` works for me. The doc didn't mention that `os.stat` attributes won't be populated on Windows. 

I agreed on file mode checking across different OS platform, but if file mode checking differentiate on Windows, should we add additional support?

Reply:
I also didn't find the doc for `os.stat`'s behavior on Windows.

`st_mode=33206` is the default for a Windows file, which is `‭1000000 110 110 110‬` in binary. You can see `rwx` is the same `110` for different scopes. However you change it, it keeps this pattern. For example, if you change the file to read-only via `os.chmod('{}', stat.S_IRUSR)`, `os.stat` is `33060` which is `‭1000000 100 100 100‬` in binary. 

Checking permission on Windows requires `win32security` according to https://stackoverflow.com/a/12168268/2199657. This will over complicate the test. Since we only save the config file under `~/.azure`, we can assume Windows behaves correctly unless the system is compromised, but of course this is out of the scope of the app itself.

Reply:
I believe you are running this command on Cygwin which has additional file management logic. Running `os.chmod` and `os.stat` on bare Windows will repro my observation.

Reply:
No, on CMD